src/firmware/app/control/python_controller/difference_equation_test.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+"""
+This file contains the unit tests for the DifferenceEquation class
+
+"""
+
+import numpy as np
+import control as ct
+from control.matlab import *
+import matplotlib.pyplot as plt
+import unittest
```

Comment:
is there any reason why we chose `unittest` over `pytest`? `pytest` makes things suuuper easy to test, each test is a function instead of a class and `pymock` is magical.

Reply:
Mat, Evan, and I had a chat about this, just seemed like overkill here, no need to add another library unless we have to. At the same time, if it's that much more fully featured, perhaps we can consider switching even if we don't need it here just so that we can be consistent with any future tests that do need those features.

Reply:
Yea I think I mis-remembered / mis-spoke oringinally. I think this is a good opportunity to at least see what it looks like and standardize early.

Reply:
Just for clarity here, @akhilveeraghanta  we're planning to try switch to `pytest` but don't want to block this PR. @EvanMorcom please make a ticket for moving to `pytest`

Reply:
Done, #1381